packages/select-rich/src/LionSelectRich.js

```diff
@@ -338,8 +327,19 @@ export class LionSelectRich extends OverlayMixin(
     /* eslint-enable no-param-reassign */
   }
 
-  _getFromAllFormElements(property) {
-    return this.formElements.map(e => e[property]);
+  __delegateNameAttribute(child) {
+    if (!child.name || child.name === this.name) {
+      // eslint-disable-next-line no-param-reassign
+      child.name = this.name;
+    } else {
+      throw new Error(
+        `The lion-select-rich name="${
+          this.name
+        }" does not allow to register ${child.tagName.toLowerCase()} with custom names (name="${
+          child.name
+        }" given)`,
+      );
+    }
```

Comment:
`__delegateNameAttribute` also exists in `ChoiceGroupMixin` with almost the same body. Can we remove it from `LionSelectRich`?

Reply:
Yes I would be in favor :) I probably missed it